mlir/lib/Util/Runner.cpp

```diff
@@ -236,14 +236,36 @@ class AIRRunner::AIRRunner_impl {
     return execution_time;
   }
 
-  void processGraph(runnerNode &c, device &device_resource_node,
+  bool processGraph(runnerNode &c, device &device_resource_node,
                     uint64_t time) {
 
+    LLVM_DEBUG(llvm::dbgs() << "\nNEW TIME STAMP @" << time - 1 << " runner "
+                            << air::to_string(c.ctrl_g->hierarchyOp) << " loc "
+                            << air::to_string(c.ctrl_g->position) << "'\n");
+
+    executeOpsFromWavefrontAndFreeResource(c, device_resource_node, time);
+    pushOpsToWavefrontAndAllocateResource(c, device_resource_node, time);
+
+    return c.wavefront.size() > 0;
```

Comment:
Perhaps this API does exist?
```suggestion
    return !c.wavefront.empty();
```